lib/runtime/wasmer/genesis.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package wasmer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/trie"
+)
+
+var (
+	ErrGenesisTopNotFound = errors.New("genesis top not found")
+)
+
+// NewTrieFromGenesis creates a new trie from the raw genesis data
+func NewTrieFromGenesis(gen genesis.Genesis) (tr *trie.Trie, err error) {
+	tr = trie.NewEmptyTrie()
+	genesisFields := gen.GenesisFields()
+	keyValues, ok := genesisFields.Raw["top"]
+	if !ok {
+		return nil, fmt.Errorf("%w: in genesis %s",
+			ErrGenesisTopNotFound, gen.Name)
+	}
+
+	err = tr.LoadFromMap(keyValues)
+	if err != nil {
```

Comment:
Not possible since the wasmer package imports the trie package, so we cannot have this in the trie package. I wish it would be possible and totally agree apart from the cycle dependency problem!

Reply:
As I mentioned above:

> Soon this will instantiate a wasmer instance from the genesis runtime code, and load the state version from it. That's why this is in the wasmer package.

So it won't work. That PR was especially made for the upcoming branch based on it (#2774) 😉